inventory/any.cf

```diff
@@ -520,38 +520,30 @@ bundle agent cfe_autorun_inventory_aws_ec2_metadata
 {
   methods:
     !(disable_inventory_aws|disable_inventory_aws_ec2_metadata)::
-      "cfe_autorun_inventory_aws_ec2_metadata_cache";
+      "cfe_autorun_inventory_aws_ec2_metadata_cache"
+        action => if_elapsed_day;
       "cfe_aws_ec2_metadata_from_cache";
 }
 bundle agent cfe_autorun_inventory_aws_ec2_metadata_cache
-# @brief Cache ec2 metadata from http request for one day
+# @brief Cache ec2 metadata from http request
+# It's up to caller to ensure this bundle is called only once a day
 #
 # Provides cache of ec2 instance metadata for inventory
 {
 
   vars:
-
     ec2_instance.!(disable_inventory_aws|disable_inventory_aws_ec2_metadata)::
-
-      "curl"      string => "$(paths.curl)";
-      "curl_opts" string => "--max-time 1 --connect-timeout 1 --silent";
       "URL"       string => "http://169.254.169.254/latest/dynamic/instance-identity/document";
       "cache"     string => "$(sys.statedir)/aws_ec2_metadata";
-
-  commands:
-
-    ec2_instance.!(disable_inventory_aws|disable_inventory_aws_ec2_metadata)::
-
-      "$(curl)"
-        args => "$(curl_opts) $(URL) -o $(cache)",
-        if => and( not( fileexists( $(cache) )),
-                   isexecutable( $(curl) ));
+      "response"  data => url_get($(URL), '{"url.max_content": 512000, "url.timeout": 1}');
+      "data"      string => "$(res[content])";
 
   files:
     ec2_instance.!(disable_inventory_aws|disable_inventory_aws_ec2_metadata)::
       "$(cache)"
-        delete => tidy,
-        file_select => days_old(1);
+        edit_line => insert_lines($(data)),
```

Comment:
Instead of using edit_line, I would probably use an inline mustache template to render the file.